ui/pages/swaps/prepare-swap-page/prepare-swap-page.js

```diff
@@ -0,0 +1,1084 @@
+import React, { useContext, useEffect, useState, useCallback } from 'react';
```

Comment:
Most of the logic here is reused from the Build Quote page: https://github.com/MetaMask/metamask-extension/blob/develop/ui/pages/swaps/build-quote/build-quote.js